src/parser/flow_lexer.ml

```diff
@@ -724,65 +767,129 @@ let token (env: Lex_env.t) lexbuf : result =
       is_tail
     ))
 
+
+  | binbigint, word ->
+    (* Numbers cannot be immediately followed by words *)
+    recover env lexbuf ~f:(fun _ _ -> match%sedlex lexbuf with
```

Comment:
I was wondering why, in your new tests, the `0b101011101nhello` example didn't have an `Unexpected token ILLEGAL` error. It's because you're ignoring the new env and lexbuf being passed into this function. This should be

```
recover env lexbuf ~f:(fun env lexbuf -> match%sedlex lexbuf with
```

Since you're using the `env` and `lexbuf` that were in scope before `recover` was called, the `env` doesn't include the error that `recover` automatically adds.

Reply:
wow, good catch